web/profiles/custom/yalesites_profile/modules/custom/ys_views_basic/src/Plugin/views/filter/NewsYearFilter.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Drupal\ys_views_basic\Plugin\views\filter;
+
+use Drupal\Core\Database\Connection;
+use Drupal\views\Plugin\views\display\DisplayPluginBase;
+use Drupal\views\Plugin\views\filter\InOperator;
+use Drupal\views\ViewExecutable;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Filter news by year.
+ *
+ * @ingroup views_filter_handlers
+ *
+ * @ViewsFilter("news_year_filter")
+ */
+class NewsYearFilter extends InOperator {
```

Comment:
We should refactor instances of the word "News" to use "Post" instead, to match the conventions already on the site.